### DISCUSSION TITLE
Mechanics

### DISCUSSION THREAD

**Discussion by julianogales** [SIGGgames/PacoMonkey#41] [Ideas]
### Movement
 - [x] **Move right**
 - [x] **Move left**
 - [ ] **Crouch:** while crouching, you can walk.
 - [x] **Walk**
 - [ ] **Run**
 - [x] **Jump:** depending on pressed time can jump further. 
 - [x] **Double jump:** resets when stepping on the ground.
 - [ ] \(Optional) **Hold:** can always hold to the wall and then resets jump but not double jump.

### Combat
- [ ] **Melee attack (punch)**
- [ ] **Ranged poop attack**
 
#### Power-ups:
- [ ] **Block attacks**
- (...)
#### Items:
- [x] **Turbo**
- (...)

**Comment by Abelitux** [comment]
We should also think if it would be interesting to step down to another level of tiles (it is usually with _double s_ or _space+s_), like in this video of the game Blasphemous:

https://github.com/SIGGgames/PacoMonkey/assets/22718557/0d26c463-6259-4b5c-9dbf-7e12143a1b8d

Also, grabbing to the corner when jumping on the edge of a platform could be interesting too:

https://github.com/SIGGgames/PacoMonkey/assets/22718557/8a6f9f8e-2fbe-4421-9ad2-fb444eea0acb

Let me know what you think.

**Reply by julianogales** [reply]
We could implement both, but I prefer just doing the **grabbing to the corner** thing.

**Comment by Aniol0012** [comment]
How do you want to implement run? Like press shift and depend on stamina or what's your take on this?

**Reply by julianogales** [reply]
I'd prefer that running could be used freely, without any stamina restriction.

**Comment by Aniol0012** [comment]
REQUEST: **Hold**: can always hold to the wall and then resets jump but not double jump.

To implement this I need to make a special block to be wall so I can distinguish between both cases. Is it realistic or we just mark it as maybe TODO? @julianogales @Abelitux
